### ISSUE TITLE
optionally prevent adding the translate fields to content_panels ?

### ISSUE THREAD

**Issue by richtier** [infoportugal/wagtail-modeltranslation#173]
is there a built-in way to prevent the [addition of the language fields](https://github.com/infoportugal/wagtail-modeltranslation/blob/e8f969a87e85ce2ec014af7a05b3b77c0b35b822/wagtail_modeltranslation/patch_wagtailadmin.py#L112) to admin `content_panels`, `settings_panels`, etc?

I wish to put a language per tab. I will handle the `InlinePanel` translations a different way.

I achieved it this way so far:

```
from modeltranslation.utils import build_localized_fieldname

from django.conf import settings

content_panel_kwargs = [
    {'field_name': 'title', 'classname': 'full'},
    {'field_name': 'lede', 'classname': 'full'},
    ...
]

def make_panel(language):
    def add_suffix(kwargs):
        field_name = build_localized_fieldname(kwargs['field_name'], language)
        return {**kwargs, 'field_name': field_name}
    return[FieldPanel(**add_suffix(kwargs)) for kwargs in content_panel_kwargs]

class MyPage(Page):
   ...
    edit_handler = TabbedInterface([   
        ObjectList(make_panel(language), heading=name)
        for language, name in settings.LANGUAGES
    ])
```

but I suspect this is a commonly requested feature, so I suspect there is an eaiser way.

I see the `InlinePanel` complicates matters, but if no InlinePanels are needed then that problem goes away.

![translations](https://user-images.githubusercontent.com/5485798/36233493-3f087c6c-11de-11e8-8cd2-864e8bf2c10c.gif)

**Comment by dmarcelino** [comment]
Hi @richtier,

That interface looks very nice! It's certainly better for translating more than a couple of languages.

Back in https://github.com/wagtail/wagtail/issues/96#issuecomment-363382069 @nifel87 requested something similar.

> is there a built-in way to prevent the [addition of the language fields](https://github.com/infoportugal/wagtail-modeltranslation/blob/e8f969a87e85ce2ec014af7a05b3b77c0b35b822/wagtail_modeltranslation/patch_wagtailadmin.py#L112) to admin `content_panels`, `settings_panels`, etc?

Not one that I'm aware of but I would be OK with that being configurable via [settings](http://wagtail-modeltranslation.readthedocs.io/en/latest/advanced%20settings.html).

IMHO it would be great if `wagtail-modeltranslation` could offer such tabbed interface as an option, much like `django-modeltranslation` does where one can select the Django Admin translation interface:  http://django-modeltranslation.readthedocs.io/en/latest/admin.html#tabbed-translation-fields-admin-classes. I could see this being done either through settings, through inheriting specific classes or perhaps through `translation.py`.

**Comment by mzoeller** [comment]
I have implemented something similar based on Javascript only using a dropdown select instead of tabs (we will have too many languages even for tabs).

![wagtail-internationalization-dropdown](https://user-images.githubusercontent.com/1606177/36619322-bc08e582-18ee-11e8-8274-7df36243a681.png)

You can just copy it into one of your template dirs as "wagtailadmin/pages/edit.html":

```
{% extends "wagtailadmin/pages/edit.html" %}
{% load i18n %}

{% block extra_js %}
    {{ block.super }}

    <script>
        var translatables = {};

        function change_language(active_language) {
            for( var key in translatables )
                translatables[key].map( function(obj) {
                    obj.toggle(key === active_language)
                });
        }

        $(function () {
            $('ul[class="objects"]').find('label:contains(" [")').each(function () {
                var label = $.trim($(this).text());
                var language = label.substr(label.lastIndexOf('[') + 1, 2);
                if( !(language in translatables))
                    translatables[language] = [];
                translatables[language].push($(this).closest('li'));
            });

            var s = $('<select id="language_select" style="padding:9px; width: auto; margin-left:8px;" />');
            for (var key in translatables)
                $('<option />', {value: key, text: key}).appendTo(s);
            $("ul[class='tab-nav merged']").append(s);

            $("#language_select").on('change', function () {
                change_language(this.value);
            });

            change_language("en");
        });
    </script>
{% endblock %}
```

**Comment by stuaxo** [comment]
Both solutions work + probably need a little polish.

The tabs, look more like other parts of wagtail.

The dropdown seems closer to a PR - in that it works without extra code in the users models and doesn't break inlinetabs.   -

It probably should look a bit more like other parts of the wagtail admin.

**Comment by stuaxo** [comment]
@mzoeller the code needs to run create.html as well as edit.html.

Combined with some css to give it square corners and flatten it, fits in a little more.

This kickstarter  https://www.kickstarter.com/projects/noripyt/wagtails-first-hatch
has a mockup combining tabs and a dropdown for when there are many lanuages, this is my favourite solution yet.

![image](https://user-images.githubusercontent.com/179677/38749103-5af15b00-3f48-11e8-8c7b-89731125d4ce.png)

**Comment by Pomax** [comment]
This looks great - honestly as a user of wagtail and the current wagtail-modeltranslation, I'd much rather this lands early so I can start using it rather than waiting for someone to get some time to put on the "finishing touches": CSS/template tweaks can always be a point release =)

**Comment by stuaxo** [comment]
@Pomax I would have a play and see if you can improve it.

It turns out a purely dropdown based version is quite painful for users (who want to switch between languages).

Tabs are fine until you have lots of languages.

The hybrid - tabs + dropdown seems to be the ideal.

The other thing missing is a way to show errors in the other languages (maybe a red underline, under the languages where there are form errors).

If you can make these or any other improvements to these, then it'll get closer to landing.


Here is some CSS makes the dropdown more square.
https://github.com/uktrade/invest/blob/aac9b0e0c1f0e1569649f23e43df4e22a1e56974/invest/static/css/admin-extras.css

I'm not working with wagtail any more, so others will need to take it further :)

**Comment by dantabel** [comment]
Has there been any movement on this in the last 18 months? This functionality would be super useful.

**Comment by DiogoMarques29** [comment]
Hi @dantabel 
No news. There is a pending PR #211 for this, but a problem was found with inlines.

**Comment by DiogoMarques29** [comment]
Closed in favor of #211
